### PR TITLE
Added more events to complete the event pipelines

### PR DESCRIPTION
**Pull request by MuskanPaliwal**
- There are a few events where we are supposed to check if user is logged-in and use his `user_id` instead `anonymous_id` but I ignored it as of now, since it seemed a tedious task.
- Will create an issue for that.

### CODE REVIEW

**Comment by tapishr** [comment]
So we will get anonymous id even when the user is logged in?

**Review by tapishr — APPROVED** [review]

**Comment by MuskanPaliwal** [comment]
> So we will get anonymous id even when the user is logged in?

Yeah

**Review by tapishr — DISMISSED** [review]

**Comment by MuskanPaliwal** [comment]
-This PR address all the issues regarding event pipeline
- Changed data-type of `userId` from `number` to `string` 
- Reflected the similar change in DB as well

**Review by tapishr — APPROVED** [review]
Please change the sql tables as well before merging so queries dont fail. If possible, test with db before merging

**Comment by MuskanPaliwal** [comment]
> Please change the sql tables as well before merging so queries dont fail. If possible, test with db before merging

I did that last night itself and ab sab kuch phatt raha hain so meri PR should be merged asap 🥲

File: views/RepoList.tsx [hunk 1]
```diff
@@ -39,13 +40,13 @@ const RepoList = (props: { repo_list: string[] }) => {
 	</>)
 }
 
-export async function getRepoList(conn: Pool, userId?: number) {
+export async function getRepoList(conn: Pool, userId?: string) {
 	const repo_list_q = `SELECT DISTINCT 
 		c.commit_json ->> 'repo_name' AS repo_name
-		FROM ` + ((!userId) ? "commits AS c" : `(SELECT 
+		FROM ` + ((!convert(userId)) ? "commits AS c" : `(SELECT
```
Comment (avikalpg) [inline views/RepoList.tsx]: unnecessary.

**Review by avikalpg — CHANGES_REQUESTED** [review]

File: components/LoginLogout.tsx [hunk 2]
```diff
@@ -26,13 +58,13 @@ export default function LoginLogout() {
...
-					<li className='border-b-2 border-b-gray-200 p-2 text-center'>
+					<li id='contribute-link' className='border-b-2 border-b-gray-200 p-2 text-center'>
 						<Link href='https://github.com/Alokit-Innovations/' target='_blank' className='cursor-pointer w-full'>Contribute</Link>
 					</li>
-					<li className='border-b-2 border-b-gray-200 p-2 text-center'>
+					<li id='settings-link' className='border-b-2 border-b-gray-200 p-2 text-center'>
 						<Link href='/settings' className='cursor-pointer w-full'>Settings</Link>
 					</li>
-					<li className='p-2 text-center cursor-pointer' onClick={() => logout()}>
+					<li id='logout-link' className='p-2 text-center cursor-pointer' onClick={() => { if (rudderEventMethods) {logout(getAuthUserId(session), getAuthUserName(session), getAndSetAnonymousIdFromLocalStorage(), rudderEventMethods)}}}>
```
Comment (avikalpg) [inline components/LoginLogout.tsx:67]: This restricts logging out if `rudderEventMethods` are not found.
Maybe don't stop a person for being able to log out.

File: components/LoginLogout.tsx [hunk 3]
```diff
@@ -42,7 +74,9 @@ export default function LoginLogout() {
 		</>
 	)
 	else return (
-		<Button variant='contained' onClick={() => login()} className="rounded bg-inherit sm:bg-primary-main text-secondary-main py-2 px-4 font-semibold">
+		<Button variant='contained' onClick={() => { if (rudderEventMethods) {
+			return login(getAndSetAnonymousIdFromLocalStorage(), rudderEventMethods);
+		  }}} className="rounded bg-inherit sm:bg-primary-main text-secondary-main py-2 px-4 font-semibold">
```
Comment (avikalpg) [inline components/LoginLogout.tsx:79]: same as above.

File: pages/docs/index.tsx [hunk 4]
```diff
@@ -11,13 +16,40 @@ const verifySetup = [
 ]
 
 const Docs = ({ bitbucket_auth_url }: { bitbucket_auth_url: string }) => {
+	const { rudderEventMethods } = React.useContext(RudderContext);
```
Comment (avikalpg) [inline pages/docs/index.tsx:19]: Please check: I think you have already done this in _app.tsx

File: pages/index.tsx [hunk 5]
```diff
@@ -1,40 +1,41 @@
 import React from 'react'
+import { Session } from 'next-auth'
```
Comment (avikalpg) [inline pages/index.tsx:2]:
```suggestion
import type { Session } from 'next-auth'
```

File: pages/index.tsx [hunk 6]
```diff
@@ -1,40 +1,41 @@
...
 import WhyUs from '../views/WhyUs'
 import Features from '../views/Features'
 import TrustUs from '../views/TrustUs'
-import { rudderEventMethods } from "../utils/rudderstack_initialize";
-import { v4 as uuidv4 } from 'uuid';
-
+import RudderContext from '../components/RudderContext'
+import { getAndSetAnonymousIdFromLocalStorage } from '../utils/url_utils'
 import JoinSlack from '../views/JoinSlack'
+import { useSession } from 'next-auth/react'
```
Comment (avikalpg) [inline pages/index.tsx:12]: remove unnecessary import.

File: utils/auth.ts [hunk 7]
```diff
@@ -1,21 +1,26 @@
...
 		window.location.href = "/";
 	})
 		.catch(err => {
+			rudderEventMethods?.track(userId, "logout", {userId: userId, eventStatusFlag: 0, source: "profile-popup", name: userName}, anonymousId)
 			console.error("[signOut] Signout failed", err);
 		})
 }
 
-export const getAuthUserId = (session: Session | null) => session?.user.id ? session.user.id : 0;
+export const getAuthUserId = (session: Session | null) => session?.user.id ? session.user.id : "0";
```
Comment (avikalpg) [inline utils/auth.ts:23]:
```suggestion
export const getAuthUserId = (session: Session | null) => session?.user.id ? session.user.id : "";
```

File: pages/docs/index.tsx [hunk 8]
```diff
@@ -11,13 +16,40 @@ const verifySetup = [
...
+	const session: Session | null = useSession().data;
+	React.useEffect(() => {
+		const anonymousId = getAndSetAnonymousIdFromLocalStorage()
+		rudderEventMethods?.track(getAuthUserId(session), "docs page", { type: "page", eventStatusFlag: 1, name: getAuthUserName(session) }, anonymousId)
+
+		const handleGitHubAppClick = () => {
+			rudderEventMethods?.track(getAuthUserId(session), "Install github app ", { type: "link", eventStatusFlag: 1, source: "docs", name: getAuthUserName(session) }, anonymousId)
```
Comment (avikalpg) [inline pages/docs/index.tsx:26]:
```suggestion
			rudderEventMethods?.track(getAuthUserId(session), "Install github app", { type: "link", eventStatusFlag: 1, source: "docs", name: getAuthUserName(session) }, anonymousId)
```

File: pages/docs/index.tsx [hunk 9]
```diff
@@ -11,13 +16,40 @@ const verifySetup = [
...
+		};
+
+		const handleAuthoriseBitbucketOauthButton = () => {
+			rudderEventMethods?.track(getAuthUserId(session), "Authorise bitbucket consumer button", { type: "button", eventStatusFlag: 1, source: "docs", name: getAuthUserName(session) }, anonymousId)
```
Comment (avikalpg) [inline pages/docs/index.tsx:30]:
```suggestion
			rudderEventMethods?.track(getAuthUserId(session), "Authorise bitbucket consumer", { type: "button", eventStatusFlag: 1, source: "docs", name: getAuthUserName(session) }, anonymousId)
```

File: pages/docs/index.tsx [hunk 10]
```diff
@@ -1,8 +1,13 @@
 import React, { useState } from 'react';
+import { useSession } from 'next-auth/react'
+import { Session } from 'next-auth'
```
Comment (avikalpg) [inline pages/docs/index.tsx:3]:
```suggestion
import type { Session } from 'next-auth'
```

File: pages/pricing.tsx [hunk 11]
```diff
@@ -1,11 +1,13 @@
 import React, { useState } from 'react'
+import { useSession } from 'next-auth/react'
+import { Session } from 'next-auth'
```
Comment (avikalpg) [inline pages/pricing.tsx:3]:
```suggestion
import type { Session } from 'next-auth'
```

File: pages/pricing.tsx [hunk 12]
```diff
@@ -61,23 +64,9 @@ const Pricing = () => {
...
-
-	React.useEffect(() => {
-		const localStorageAnonymousId = localStorage.getItem('AnonymousId');
-		const anonymousId: string = (localStorageAnonymousId && localStorageAnonymousId != null) ? localStorageAnonymousId : uuidv4();
-		rudderEventMethods().then((response) => {
-			response?.track("", "pricing-page", { eventStatusFlag: 1 }, anonymousId) //Anonymous Id is set in local storage as soon as the user lands on the webiste.
-		});
-	}, []);
+		const anonymousId = getAndSetAnonymousIdFromLocalStorage()
+		rudderEventMethods?.track(getAuthUserId(session), "pricing-page", { type: "page", eventStatusFlag: 1, name: getAuthUserName(session) }, anonymousId) //Anonymous Id is set in local storage as soon as the user lands on the webiste.
```
Comment (avikalpg) [inline pages/pricing.tsx:68]:
```suggestion
		rudderEventMethods?.track(getAuthUserId(session), "pricing-page", { type: "page", eventStatusFlag: 1, name: getAuthUserName(session) }, anonymousId) //Anonymous Id is set in local storage as soon as the user lands on the website.
```

Comment (avikalpg) [reply pages/pricing.tsx:68]:
```suggestion
		rudderEventMethods?.track(getAuthUserId(session), "pricing-page", { type: "page", eventStatusFlag: 1, name: getAuthUserName(session) }, anonymousId)
```

File: pages/u.tsx [hunk 13]
```diff
@@ -21,18 +22,29 @@ type ProfileProps = {
...
 const Profile = ({ repo_list }: ProfileProps) => {
 	const session: Session | null = useSession().data;
+	const { rudderEventMethods } = useContext(RudderContext);
 	useEffect(() => {
-		rudderEventMethods().then((response) => {
-			response?.track(`${getAuthUserId(session)}`, "Repo Profile Page", { "userId": `${getAuthUserId(session)}`, "name": getAuthUserName(session) }, `${null}`);
-		});
-	}, [session])
+		const anonymousId = getAndSetAnonymousIdFromLocalStorage()
+		rudderEventMethods?.track(getAuthUserId(session), "Repo Profile Page", { type: "page", name: getAuthUserName(session) }, anonymousId);
```
Comment (avikalpg) [inline pages/u.tsx:28]:
```suggestion
		rudderEventMethods?.track(getAuthUserId(session), "User Profile Page", { type: "page", name: getAuthUserName(session) }, anonymousId);
```

File: views/JoinSlack.tsx [hunk 14]
```diff
@@ -1,8 +1,30 @@
...
+import { getAuthUserId, getAuthUserName } from '../utils/auth'
 
 const JoinSlack = () => {
+	const { rudderEventMethods } = React.useContext(RudderContext);
+	const session: Session | null = useSession().data;
+	React.useEffect(() => {
+		const anonymousId = getAndSetAnonymousIdFromLocalStorage()
+
+		const handleJoinSlack = () => {
+			rudderEventMethods?.track(getAuthUserId(session), "join slack", { type: "button", eventStatusFlag: 1, name: getAuthUserName(session) }, anonymousId)
```
Comment (avikalpg) [inline views/JoinSlack.tsx:17]: add `source: "landing page banner"`

**Review by avikalpg — APPROVED** [review]

**Pull request merged**